packages/react-components/src/components/VideoGallery/useVideoStreamLifecycleMaintainer.ts

```diff
@@ -53,36 +57,47 @@ const useVideoStreamLifecycleMaintainer = (props: VideoStreamLifecycleMaintainer
   // that recreates the stream view.
 
   const [streamRendererResult, setStreamRendererResult] = React.useState<CreateVideoStreamViewResult>();
-  const scalingModeRef = useRef(scalingMode);
-  const hasScalingModeChanged = scalingModeRef.current !== scalingMode;
-  const updatingScalingModeDirectly = hasScalingModeChanged && !!streamRendererResult;
+  const scalingModeForDirectUpdateRef = useRef(scalingMode);
+  const hasScalingModeChanged = scalingModeForDirectUpdateRef.current !== scalingMode;
+  const canUpdateScalingModeDirectly = !!streamRendererResult?.view.updateScalingMode;
 
   const [triggerRescale, cancelRescale] = useCancellableTask();
   const [triggerCreateStreamView, cancelCreateStreamView] = useCancellableTask();
 
-  if (isStreamAvailable && renderElementExists && scalingMode && updatingScalingModeDirectly) {
+  if (
+    isStreamAvailable &&
+    renderElementExists &&
+    scalingMode &&
+    hasScalingModeChanged &&
+    canUpdateScalingModeDirectly
+  ) {
     triggerRescale(async (cancellable: Cancellable) => {
       streamRendererResult && (await streamRendererResult.view.updateScalingMode(scalingMode));
       if (cancellable.cancelled) {
         return;
       }
-      scalingModeRef.current = scalingMode;
+      scalingModeForDirectUpdateRef.current = scalingMode;
     });
   }
 
   // scalingModeForUseEffect will trigger the useEffect to recreate the stream view only if the scaling mode has changed and
   // we cannot call updateScalingMode on the view object directly. Otherwise it will be null to ensure the useEffect does not trigger
   // when scaling mode changes.
-  const scalingModeForUseEffect = updatingScalingModeDirectly ? null : scalingMode;
```

Comment:
This was ultimately what what was wrong:
* On first render, scaling mode is "Crop", therefore the dep in the useEffect below is Crop
* When we finish `onCreateStreamView` and this has another render pass with RenderElement === true, `updatingScalingModeDirectly` can also be true now
* When `updatingScalingModeDirectly` the scalingModeForUseEffect is now `null`
  * _This now differs from "Crop" it originally was causing the below useEffect to destroy and recreate the stream_